distributed/tests/test_steal.py

```diff
@@ -1703,9 +1686,12 @@ async def _dependency_balance_test_permutation(
         workers,
     )
 
-    if recompute_saturation:
-        for ws in s.workers.values():
-            s._reevaluate_occupancy_worker(ws)
+    # Re-evaluate idle/saturated classification to avoid outdated classifications due to
+    # the initialization order of workers. On a real cluster, this would get constantly
+    # updated by tasks completing (except for stragglers).
```

Comment:
Hm, as I mentioned in the comment above, there are definitely use-cases right now where people submit very, very slow tasks (~hours) and expect them to rebalance to new workers, even before any tasks have completed.

Adding this re-evaluation in here to every test case might hide things that could cause actual problems in this case?

Reply:
I've adjusted the comment. The idle/saturated classification happens whenever a task completes **or is added**.

In your example, if we assume tasks being scheduled _roughly_ round-robin, all workers should end up as saturated in the beginning. Since new workers would be idle, stealing should happen. 

The problem in this particular test _implementation_ is that we do not schedule tasks round-robin but for one worker after another. Thus, if we first schedule all specified tasks on a worker that's supposed to be idle in the grand scheme of the test, it would be classified as saturated for the lack of other _even more_ saturated workers. These other workers only get filled up after this worker, which requires us to reclassify once after we completed the test setup.